src/modules/top-level.nix

```diff
@@ -175,12 +175,16 @@ in
         # - free to create as an unprivileged user across OSes
         default =
           let
+            runtimeEnv = builtins.getEnv "DEVENV_RUNTIME";
```

Comment:
I'm not following, what sets `$DEVENV_RUNTIME` when evaluating using `nix develop`?

Reply:
We first eval the shell and compute the initial `devenv.runtime` based on the either the `XDG_RUNTIME_DIR` or `TMPDIR`.  This then becomes `$DEVENV_RUNTIME` in the launched shell.

In the shell, `nix develop` updates the `TMPDIR` env, but `devenv.runtime` has no way of picking this up.

Once inside the shell, we run `devenv up`, which re-evaluates the `procfileScript` and picks up the new `TMPDIR`. `devenv.runtime` is now different, but the `enterShell` script that creates the runtime dir is not run again.


This is a bit of a hack anyways. I think the right way to solve this is to eval everything once. Something like #1142 should fix this.